DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/db_utils/LikeRegex.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef __LIKEREGEX_HPP__
+#define __LIKEREGEX_HPP__
+#pragma once
+
+#include <array>
+#include <type_traits>
+
+//
+// Regex LIKE engine that can match: *WORD*, *WORD and WORD*
+// where the string has MaxStrLength words or length at most MaxWordLength.
+// If words are shorter than MaxWordLength, they are padded with '\0'.
+//
+template <unsigned int MaxWordLength, unsigned int MaxStrLength>
```

Comment:
What is our preference for template parameters? the 'k' prefix seems weird too me. But so does snake case...

Reply:
Only thing I see in the Google standard is them using 'T' for a template parameter (uppercase). I followed that lead.

Reply:
When the template parameters are a class/typename it's OK to use MixedCase.   The question was when it's a value.

Here to follow the style guideline it seems it should be

template\<uinsigned int max_word_length, unsigned int max_string_length\>

Reply:
So if you have a mix, it will look like this:
template<typname MyType, int my_integer, typename MyOtherType, int my_other_integer>

Is this acceptable?

Reply:
I have followed your suggestions with the recent code.

Reply:
While it looks a bit peculiar, I actually rather like the convention -- makes it easy to visually differentiate templated types from templated values.